src/stdune/path.ml

```diff
@@ -468,12 +468,16 @@ let insert_after_build_dir_exn =
       ]
   in
   fun a b ->
-    if not (is_local a) || String.contains b '/' then error a b;
-    match String.lsplit2 a ~on:'/' with
-    | Some ("_build", rest) ->
-      Printf.sprintf "_build/%s/%s" b rest
-    | _ ->
+    if not (is_local a) then
       error a b
+    else if a = build_dir then
+      relative build_dir b
+    else
+      match String.lsplit2 a ~on:'/' with
+      | Some (build_dir', rest) when build_dir = build_dir' ->
+        relative build_dir (Printf.sprintf "%s/%s" b rest)
```

Comment:
At this point we know that `rest` is already a valid local path. So it would be more efficient to write the code this way: `Local.append (relative build_dir b) rest`. This way we avoid re-scanning and splitting `rest`

Reply:
Ah yes, fixed. Note that in the symlink branch, all this will not be required anyway.

Reply:
ack